group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -5524,5 +5670,2897 @@ private static Record newGroupMetadataRecord(
+            .withMemberId(leaderId)
+            .withProtocolType(protocolType.orElse(null))
+            .withProtocolName(protocolName.orElse(null))
+            .withGenerationId(generationId)
+            .withAssignment(assignment)
+            .build();
+
+        CompletableFuture<SyncGroupResponseData> leaderSyncFuture = new CompletableFuture<>();
+        result = context.sendGenericGroupSync(syncRequest, leaderSyncFuture);
+
+        if (expectedError == Errors.NONE) {
+            assertEquals(
+                Collections.singletonList(newGroupMetadataRecord(group, MetadataVersion.latest())),
+                result.records()
+            );
+            // Simulate successful write to log.
+            result.appendFuture().complete(null);
+        } else {
+            assertTrue(result.records().isEmpty());
+        }
+
+        assertTrue(leaderSyncFuture.isDone());
+        assertEquals(expectedError.code(), leaderSyncFuture.get().errorCode());
+        assertEquals(expectedProtocolType.orElse(null), leaderSyncFuture.get().protocolType());
+        assertEquals(expectedProtocolName.orElse(null), leaderSyncFuture.get().protocolName());
+
+        CompletableFuture<SyncGroupResponseData> followerSyncFuture = new CompletableFuture<>();
+        result = context.sendGenericGroupSync(syncRequest.setMemberId(followerId), followerSyncFuture);
+
+        assertTrue(result.records().isEmpty());
+        assertTrue(followerSyncFuture.isDone());
+        assertEquals(expectedError.code(), followerSyncFuture.get().errorCode());
+        assertEquals(expectedProtocolType.orElse(null), followerSyncFuture.get().protocolType());
+        assertEquals(expectedProtocolName.orElse(null), followerSyncFuture.get().protocolName());
+    }
+
+    @Test
+    public void testSyncGroupFromUnknownGroup() throws Exception {
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .build();
+
+        // SyncGroup with the provided Protocol Type and Name
+        SyncGroupRequestData syncRequest = new SyncGroupRequestBuilder()
+            .withGroupId("group-id")
+            .withMemberId("member-id")
+            .withGenerationId(1)
+            .build();
+
+        CompletableFuture<SyncGroupResponseData> syncFuture = new CompletableFuture<>();
+        CoordinatorResult<Void, Record> result = context.sendGenericGroupSync(syncRequest, syncFuture);
+
+        assertTrue(result.records().isEmpty());
+        assertTrue(syncFuture.isDone());
+        assertEquals(Errors.UNKNOWN_MEMBER_ID.code(), syncFuture.get().errorCode());
+    }
+
+    @Test
+    public void testSyncGroupFromUnknownMember() throws Exception {
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .build();
+        GenericGroup group = context.createGenericGroup("group-id");
+
+        JoinGroupRequestData joinRequest = new JoinGroupRequestBuilder()
```

Comment:
Here we do a join/sync with one member to setup the test. I wonder if it worth having an helper method in the context for this. What do you think?